src/main/java/io/ebeaninternal/server/type/ScalarTypeBaseDateTime.java

```diff
@@ -105,7 +105,9 @@ public T jsonRead(JsonParser parser) throws IOException {
       }
       default: {
         String jsonDateTime = parser.getText();
-        return convertFromTimestamp(dateTimeParser.parse(jsonDateTime));
+        Instant instant = Instant.parse(jsonDateTime.replace(" ", "T"));
+        Timestamp ts = Timestamp.from(instant);
+        return convertFromTimestamp(ts);
```

Comment:
I'm not happy with this change
1. why do you need to replace ` ` with `T`? (btw when replacing single characters it is better to use `replace(' ', 'T')`)
2. the change makes the dateTimeParser.parse method obsolete. So it would be better to change the 'parse' implementation.

What was the problem here? I assume, the precision of fractions of the DateTimeParser. Newer java versions (11+) probable needs up to 9 digits: `yyyy-MM-dd'T'HH:mm:ss.SSSSSSSSS'Z'
@rbygrave I think it would be a good idea to refactor the DateTimeParser, so that it uses Instant.parse instead of SimpleDateFormat

Reply:
> good idea to refactor the DateTimeParser, so that it uses Instant.parse instead of SimpleDateFormat

Yes, I agree @rPraml  @tobias-  ... Refactoring UtilDateTimeParser to use Instant.parse() looks like the good option.

Reply:
@rPraml @rbygrave Sorry, I managed to put extra stuff on this branch (it's been force pushed away not long after @rPraml commented on this PR). The java.time issue (#1725) is PR #1724 and nothing else. This PR is about test cases for circular dependencies.